File: src/Atlas/Battlenet/Platform.cs
namespace Atlas.Battlenet
{

    public class Platform
    {
        public enum PlatformCode
        {
            MacOSClassic = 0x504D4143, // PMAC
            MacOSX = 0x584D4143, // XMAC
            Windows = 0x49583836, // IX86
        }

        public static string PlatformName(PlatformCode code, bool extended = true)
        {
            return code switch
            {
                PlatformCode.MacOSClassic => "Mac OS Classic",
                PlatformCode.MacOSX       => "Mac OS X",
                PlatformCode.Windows      => "Windows",
                _ => "Unknown" + (extended ? " (" + code.ToString() + ")" : ""),
            };
        }
    }
}

File: src/AtlasV/Battlenet/BNET/Platform.vb
Imports System

Namespace AtlasV.Battlenet
    Public Class Platform
        Public Enum PlatformCode As UInt32
            None = 0UI
            MacOSClassic = &H504D4143UI
            MacOSX = &H584D4143UI
            Windows = &H49583836UI
        End Enum

#Disable Warning IDE0060 ' Remove unused parameter
        Public Shared Function PlatformName(ByVal code As PlatformCode, ByVal Optional extended As Boolean = True) As String
#Enable Warning IDE0060 ' Remove unused parameter
            Select Case code
                Case PlatformCode.None : Return "None"
                Case PlatformCode.MacOSClassic : Return "Mac OS Classic"
                Case PlatformCode.MacOSX : Return "Mac OS X"
                Case PlatformCode.Windows : Return "Windows"
                Case Else : Return "Unknown"
            End Select
        End Function
    End Class
End Namespace